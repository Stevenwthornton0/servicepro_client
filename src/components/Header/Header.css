.header {
    width: 100%;
    background-color: #E0E0E0;
}

.header_container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 90%;
    margin-left: auto;
    margin-right: auto;
    height: fit-content;
}

.logo_container {
    display: flex;
    flex-direction: row;
    text-decoration: none;
    width: fit-content;
    height: inherit;
    margin-left: 20px;
}


.logo1 {
    padding-top: 15px;
    padding-right: 3px;
    font-size: 20px;
    color: #424242;
    margin-top: 10px;
    margin-bottom: 10px;
    height: fit-content;
}

.logo2 {
    padding: 15px 6px;
    background-color: lightblue;
    border-radius: 100%;
    color: white;
    font-size: 20px;
    font-weight: 800;
    margin-top: 10px;
    margin-bottom: 10px;
    height: fit-content;
}

.header_logged-in {
    width: 35%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    font-size: 18px;
    color: #424242;
}

.header_logged-in > a {
    text-decoration: none;
    padding-left: 20px;
    padding-right: 20px;
}

.header_logged-in > a  > p {
    margin-top: 24px;
    color: #424242;
    font-weight: 500;
}

.header_logged-out {
    width: 100px;
    font-size: 20px;
    text-decoration: none;
    color: #424242;
}

.header_logged-out > p {
    margin: 24px auto 24px auto;
    width: fit-content;
}

.burger {
    width: 70px;
    margin-top: 25px;
}

.burger > div {
    display: block;
    border-top: 2px solid black;
    width: 30px;
    margin-bottom: 8px;
    margin-right: auto;
    margin-left: auto;
}

.menu {
    display: flex;
    flex-direction: column;
    position: absolute;
    top: 74px;
    right: 0;
    border-bottom: 1px solid #bdbdbd;
    border-right: 1px solid #bdbdbd;
    border-left: 1px solid #bdbdbd;
}

.menu > a {
    font-size: 18px;
    text-decoration: none;
    color: black;
    border-top: 1px solid #bdbdbd;
    padding: 16px;
    background-color: #e0e0e0;
}

@media screen and (min-width: 770px) {

    .header_container {
        width: 68%;
    }

    .logo_container {
        margin-left: 0;
    }

    .header_logged-in > a:hover {
        background-color: lightblue;
    }

    .header_logged-out:hover {
        background-color: lightblue;
    }

    .header_logged-out:hover {
        background-color: #e0e0e0;
    }
    
    .header_logged-in > a:hover {
        background-color: #e0e0e0;
    }
}