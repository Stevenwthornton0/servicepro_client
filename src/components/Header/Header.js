import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';
import ServicesContext from '../../contexts/ServicesContext';
import './Header.css';

class Header extends Component {

    static contextType = ServicesContext;

    handleLogoutClick = () => {
        TokenService.clearAuthToken()
        window.location.reload()
    }

    renderMenu () {
        const { open } = this.context;
        if (open) {
            return (
                <div className='menu'>
                    <Link to='/provider/registration'>Provider Registration</Link>
                    <Link
                        onClick={this.handleLogoutClick}
                        to={'/'}
                    >
                        Logout
                    </Link>
                </div>
            )
        }
    }

    renderHam () {
        return (
            <div className='burger mobile' onClick={e => this.context.setOpen(e)}> 
                <div></div>
                <div></div>
                <div></div>
            </div>
        )
    }

    renderLogoutLink() {
        return (
            <div className='header_logged-in computer'>
                <Link to='/provider/registration'>
                    <p>Provider Registration</p>
                </Link>   

                <Link
                    onClick={this.handleLogoutClick}
                    to='/'>
                    <p>Logout</p>
                </Link>
            </div>
        )
    }

    renderLoginLink() {
        return (
            <Link to={'/login'} className='header_logged-out'>
                <p>Login</p>
            </Link>
        ) 
    }

    nonUserLogo() {
        return (
            <Link to={'/'} className='logo_container'>
                <h1 className='logo1'>Service</h1>
                <h1 className='logo2'>PRO</h1>
            </Link>
        )
    }

    userLogo() {
        return (
            <Link to={'/services'} className='logo_container'>
                <h1 className='logo1'>Service</h1>
                <h1 className='logo2'>PRO</h1>
            </Link>
        )
    }

    render() {
        return (
            <div className='header'>
                <nav className='header_container'>
                    {TokenService.hasAuthToken()
                        ? this.userLogo()
                        : this.nonUserLogo()}

                    {TokenService.hasAuthToken()
                        ? this.renderLogoutLink()
                        : this.renderLoginLink()}

                    {TokenService.hasAuthToken()
                        ? this.renderHam()
                        : null}
                        
                    {this.renderMenu()}
                </nav>
            </div>
        )
    }
}

export default Header;