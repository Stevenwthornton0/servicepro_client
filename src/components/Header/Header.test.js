import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import Header from './Header';

Enzyme.configure({ adapter: new Adapter() })

describe('Header component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Header />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})