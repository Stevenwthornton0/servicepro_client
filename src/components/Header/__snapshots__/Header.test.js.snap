// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders without crashing 1`] = `
<div
  className="header"
>
  <nav
    className="header_container"
  >
    <Link
      className="logo_container"
      to="/"
    >
      <h1
        className="logo1"
      >
        Service
      </h1>
      <h1
        className="logo2"
      >
        PRO
      </h1>
    </Link>
    <Link
      className="header_logged-out"
      to="/login"
    >
      <p>
        Login
      </p>
    </Link>
  </nav>
</div>
`;
